dipy/core/gradients.py

```diff
@@ -326,11 +402,14 @@ def gradient_table_from_bvals_bvecs(bvals, bvecs, b0_threshold=50, atol=1e-2,
     if b0_threshold >= 200:
         warn("b0_threshold has a value > 199")
 
-    # checking for the correctness of bvals
-    if b0_threshold < bvals.min():
-        warn("b0_threshold (value: {0}) is too low, increase your "
-             "b0_threshold. It should be higher than the lowest b0 value "
-             "({1}).".format(b0_threshold, bvals.min()))
+    # If all b-values are smaller or equal to the b0 threshold, it is assumed
+    # that no thresholding is requested
+    if any(mask_non_weighted_bvals(bvals, b0_threshold)):
+        # checking for the correctness of bvals
+        if b0_threshold < bvals.min():
+            warn("b0_threshold (value: {0}) is too low, increase your "
+                 "b0_threshold. It should be higher than the lowest b0 value "
+                 "({1}).".format(b0_threshold, bvals.min()))
```

Comment:
Can you change this using f-string? thanks

Also, why you did not create a message like above?

Reply:
That is a separate commit and a separate PR: it does not help addressing the warnings.

Reply:
I recommend you to look at the Boyscout rule, since you edited this line, it cost nothing to clean/slightly improve at the same times